http/beaconstate.go

```diff
@@ -70,31 +71,66 @@ func (s *Service) beaconStateFromSSZ(res *httpResponse) (*api.Response[*spec.Ver
 		Metadata: metadataFromHeaders(res.headers),
 	}
 
+	var dynSSZ *dynssz.DynSsz
+	if s.useDynamicSSZ {
+		dynSSZ = dynssz.NewDynSsz(s.spec)
```

Comment:
There are no guarantees that `s.spec` is populated, this should fetch the spec via `Spec()`

Reply:
fixed